presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Memo.java

```diff
@@ -263,4 +275,59 @@ private Group(PlanNode member)
             this.membership = requireNonNull(member, "member is null");
         }
     }
+
+    private static class PlanIteratorImpl
```

Comment:
nit: Presto doesn't seem to use `XXXImpl` style class name, although it's quite popular in Hive/Spark

Reply:
XXXImpl comes from the enterprise worlds. JavaEE technologies heavily rely on proxy mechanisms for automatic transaction management and Aspect oriented filters. In the old days it wasn't possible (was problematic) to proxify classes. That's why that was a standard pattern de-facto to have `interfaces WhateverService ` and `class WhateverServiceImpl `.